Spanner/tests/Unit/SpannerClientGcpOptimizerEnabledTest.php

```diff
@@ -0,0 +1,310 @@
+<?php
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Spanner\Tests\Unit;
+
+require_once (__DIR__ . "/../../../vendor/autoload.php");
+
+use Google\Cloud\Core\Int64;
+use Google\Cloud\Core\Iterator\ItemIterator;
+use Google\Cloud\Core\LongRunning\LongRunningOperation;
+use Google\Cloud\Core\Testing\GrpcTestTrait;
+use Google\Cloud\Spanner\Admin\Database\V1\DatabaseAdminClient;
+use Google\Cloud\Spanner\Admin\Instance\V1\InstanceAdminClient;
+use Google\Cloud\Spanner\Batch\BatchClient;
+use Google\Cloud\Spanner\Bytes;
+use Google\Cloud\Spanner\CommitTimestamp;
+use Google\Cloud\Spanner\Connection\ConnectionInterface;
+use Google\Cloud\Spanner\Database;
+use Google\Cloud\Spanner\Date;
+use Google\Cloud\Spanner\Duration;
+use Google\Cloud\Spanner\Instance;
+use Google\Cloud\Spanner\InstanceConfiguration;
+use Google\Cloud\Spanner\KeyRange;
+use Google\Cloud\Spanner\KeySet;
+use Google\Cloud\Spanner\SpannerClient;
+use Google\Cloud\Spanner\Timestamp;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+
+/**
+ * @group spanner
+ */
+class SpannerClientGcpOptimizerEnabledTest extends TestCase
```

Comment:
I'm not sure this test is doing anything useful. All the service calls in this test are mocked, which means none of these tests are actually getting to the grpc layer. We could add unit tests to `Google\Cloud\Spanner\Connection\Grpc` to test that the grpc-gcp-php integration is working correctly, but to test the actual calls, we'll need to augment our system tests.

If I'm misunderstanding the intent, please forgive me!

Reply:
I was using this test to verify that with or without the grpc-gcp library, the behavior remains the same for the spanner client usages, so no user will be blocked if we miss something.

Reply:
Thinking about it, I still doubt it'll make a difference. We should add a single unit test to verify that enabling the optimizer doesn't cause problems in instantiating the client, but other than that I don't think this is necessary.

I'll follow up afterwards with some changes to the system tests which can test the optimizer in real service calls. :)

Reply:
Hi @jdpedrie ,

I deleted this file. We have tests creating the spanner client under the `grpc-gcp-php` repo with the `grpc-gcp` enabled. It makes calls with the real service. Will that work? Other than tests, are there something else blocking merge this PR?

Thanks!